corehq/apps/app_manager/util.py

```diff
@@ -426,7 +426,7 @@ def get_cloudcare_session_data(domain_name, form, couch_user):
 
     datums = EntriesHelper.get_new_case_id_datums_meta(form)
     session_data = {datum.id: uuid.uuid4().hex for datum in datums}
-    if couch_user.doc_type == 'CommCareUser':  # smsforms.app.start_session could pass a CommCareCase
+    if couch_user.doc_type in ('CommCareUser', 'WebUser'):  # smsforms.app.start_session could pass a CommCareCase
```

Comment:
This will fail hard a few lines below - `get_usercase_id` is only defined on `CommCareUser`

Reply:
Okay went ahead and added some logic and helper functions to handle this

Reply:
I would recommend making use of the `OTARestoreUser` wrapper classes (see `corehq.apps.ota.utils.get_restore_user`). These already exists to try and abstract differences between the two so it would be good to keep that up.

Reply:
Hey @snopoke I know it's been a hot minute since this review. Can you give a bit more detail how you mean from the `OTARestoreUser` wrapper classes to be used in this instance?

Reply:
instead of having `if couch_user.doc_type == "CommCareUser": .... else ...` you could add a method to the `OTARestoreCommCareUser` and `OTARestoreWebUser` classes to get the user case ID:

```python
restore_user = couch_user.to_ota_restore_user()
usercase_id = restore_user.get_usercase_id()
```

These classes are designed to abstract the differences between web and mobile users for the restore code so I think it makes sense to use them here.

Reply:
Okay that makes a lot of sense. How does something like this look?
```python3
args = {'domain': domain_name} if couch_user.doc_type == 'WebUser' else {}
if EntriesHelper.any_usercase_datums(extra_datums):
    restore_user = couch_user.to_ota_restore_user(**args)
    usercase_id = restore_user.get_usercase_id()
    if usercase_id:
        session_data[USERCASE_ID] = usercase_id
```

Reply:
Yes, that's exactly right. It's quite irritating that the `to_ota_restore_user` has different args between the users. We could standardise that quite easily though which would make your code even cleaner. Leave the web user one as it is and update the mobile user one to:
```python
def to_ota_restore_user(self, domain, request_user=None):
    assert domain == self.domain
    ... etc
```

Reply:
Changed!

Reply:
Thanks. FYI I updated other usages of the function to pass in the domain as well